src/btree/bt_split.c

```diff
@@ -773,7 +773,8 @@ __split_multi_inmem(
 	 * transaction to an impossibly old value so this page is never skipped
 	 * in a checkpoint.
 	 */
-	page->modify->first_dirty_txn = WT_TXN_FIRST;
+	if (page->modify != NULL)
+		page->modify->first_dirty_txn = WT_TXN_FIRST;
```

Comment:
@agorrod, I don't understand why this change would be necessary; how can we be here if there's no `page->modify` structure?